tests/test_docker_cat.py

```diff
@@ -0,0 +1,539 @@
+            qp_key = requests.get(f"{cls.CAT_URL}/api/qualityprofiles/search?quality_profile={quality_profile}",
+                auth=("admin", "admin")).json()['profiles'][0]['key']
+            # Activate the rule in the Quality Profile to allow the Sensor to be used
+            requests.post(f"{cls.CAT_URL}/api/qualityprofiles/activate_rule",
+                auth=("admin", "admin"),
+                data={
+                    "key": qp_key,
+                    "rule": rule_violated
+                })
+        # Create a project on SonarQube
+        errors = requests.post(f"{cls.CAT_URL}/api/projects/create",
+            auth=("admin", "admin"),
+            data={
+                "name": project_name,
+                "project": project_key
+            }).json().get('errors', [])
+        assert not errors
+        # Set its Quality Profile for the given language to the given one
+        requests.post(f"{cls.CAT_URL}/api/qualityprofiles/add_project",
+            auth=("admin", "admin"),
+            data={
+                "language": language_key,
+                "project": project_key,
+                "qualityProfile": quality_profile
+            })
+        # Analyse the project and collect the analysis files (that match the default names)
+        docker_client = docker.from_env()
+        analysis_output = docker_client.containers.get(cls.CAT_CONTAINER_NAME).exec_run(
+            f"sonar-scanner -Dsonar.host.url=http://localhost:9000 -Dsonar.projectKey={project_key} -Dsonar.projectName=\"{project_name}\" -Dsonar.projectVersion=1.0 -Dsonar.sources={source_folder}",
+            workdir=f"/media/sf_Shared/{language_folder}").output.decode("utf-8")
+        for line in (expected_sensor, expected_import):
+            # Hint: if this test fails, the sensor for the tool or for the importation was not launched
+            assert line in analysis_output
+        # Wait for SonarQube to process the results
+        time.sleep(10)
+        # Check that the issue was added to the project
+        issues = requests.get(f"{cls.CAT_URL}/api/issues/search?componentKeys={project_key}",
+            auth=("admin", "admin")).json()['issues']
+        nb_issues = len([ issue for issue in issues if issue['rule'] == rule_violated ])
+        # Hint: an issue must be raised by the rule violated
+        assert nb_issues == 1
+        # Delete the project
+        requests.post(f"{cls.CAT_URL}/api/projects/delete",
+            auth=("admin", "admin"),
+            data={"project": project_key})
+        if activate_rule:
+            # Deactivate the rule in the Quality Profile
+            requests.post(f"{cls.CAT_URL}/api/qualityprofiles/deactivate_rule",
+                auth=("admin", "admin"),
+                data={
+                    "key": qp_key,
+                    "rule": rule_violated
+                })
+
+    @classmethod
+    def setup_class(cls):
+        """Set up the tests
```

Comment:
Codacy found an issue: [1 blank line required between summary line and description (found 0) (D205)](https://app.codacy.com/gh/cnescatlab/docker-cat/pullRequest?prid=6253839)